netsim/providers/__init__.py

```diff
@@ -55,10 +54,23 @@ def get_template_path(self) -> str:
     return 'templates/provider/' + self.provider
 
   def get_full_template_path(self) -> str:
-    return str(_files.get_moddir()) + '/' + self.get_template_path()
+    return os.path.join(str(_files.get_moddir()), self.get_template_path())
 
   def find_extra_template(self, node: Box, fname: str, topology: Box) -> typing.Optional[str]:
-    if fname in node.get('config',[]):                    # Are we dealing with extra-config template?
+    """Find template with caching to avoid repeated filesystem searches"""
```

Comment:
Nice one ;)

However, unfortunately I don't think it's useful. You're correct in assuming the template MIGHT depend on node name and other things, so the cache key MUST include the node name. That makes cache hits highly unlikely -- we'd have to render the same file twice for the same node